### ISSUE TITLE
[feature request] [proposal] Always track CER in addition to WER for speech recognition tasks

### ISSUE THREAD

**Issue by vadimkantorov** [NVIDIA/NeMo#1517]
Currently one has to choose whether `use_cer` or not, and the choice is exclusive. Both are useful as the target metric, CER is more smooth.

**Comment by rbracco** [comment]
I support this change. Can you share code for how you validate using CER? I'm using the validate code from the ASR with NeMo tutorial and it does something like 

```
for test_batch in pb(model.test_dataloader()):
          test_batch = [x.cuda() for x in test_batch]
          xs, x_lens, ys, y_lens = test_batch
          log_probs, encoded_len, greedy_predictions = model(
              input_signal=xs, input_signal_length=x_lens)
          # Notice the model has a helper object to compute WER
          model._wer.update(greedy_predictions, ys, y_lens)
          _, wer_num, wer_denom = quartznet._wer.compute()
          wer_nums.append(wer_num.detach().cpu().numpy())
          wer_denoms.append(wer_denom.detach().cpu().numpy())
```

and since `_WER` is already instantiated in quartznet I'm not sure how to turn CER on.

**Comment by vadimkantorov** [comment]
I think you need to pass `use_cer = True` to `_wer` constructor

**Comment by rbracco** [comment]
I figured this out and will note it here in case anyone wants to use CER in the future. Note `quartznet` is the variable name of my model so you may need to change it if your model has a different name. 

The easiest way is to toggle between CER and WER by setting `quartznet._wer.use_cer`,  
`quartznet._wer.use_cer=True` will switch it to use CER, `quartznet._wer.use_cer=False`  will switch it back to WER. 

Here is what the constructor looks like for WER and how you can override
```
from nemo.collections.asr.metrics.wer import WER
quartznet._wer = WER(
            vocabulary=quartznet.decoder.vocabulary,
            batch_dim_index=0,
            use_cer=True,
            ctc_decode=True,
            dist_sync_on_step=True,
            log_prediction=True,
        )
```

Since `log_every_n_steps` doesnt appear to apply to validation as discussed in #1518, you can use this to turn off the output logs if printing every single line is causing problems. This can also be done directly with `quartznet._wer.log_prediction=False` 

Hope this helps someone!

**Comment by titu1994** [comment]
Since .metrics are calculates on cpu we prefer to preserve only one metric during training. Offline it is much easier to evaluate models with large batch sizes as well.
